src/main/java/seedu/address/logic/parser/FilterCommandParser.java

```diff
@@ -0,0 +1,109 @@
+            }
+            PersonWithCriteriaPredicate criteria = new PersonWithCriteriaPredicate(ranges);
+            return new FilterCommand(criteria);
+        } else {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FilterCommand.MESSAGE_USAGE));
+        }
+    }
+
+    /**
+     * Parses a range in string into Range object with LocalDate type.
+     * The input string must be in the format "dd/MM/yyyy - dd/MM/yyyy".
+     *
+     * @param trimmedRange the string representing the date range, expected in the format "dd/MM/yyyy - dd/MM/yyyy".
+     * @return a {@code Range<LocalDate>} object containing the parsed start and end dates.
+     * @throws ParseException if the input string is not in the correct format or contains invalid dates.
+     */
+    private Range<LocalDate> parseAppointmentRange(String trimmedRange) throws ParseException {
+        String[] splittedRange = trimmedRange.split("-");
+
+        if (splittedRange.length != 2) {
+            throw new ParseException("Incorrect range format: Please provide your appointment dates range as "
+                   + "dd/MM/yyyy - dd/MM/yyyy");
+        }
+        try {
+            LocalDate lower = LocalDate.parse(splittedRange[0].trim(), FORMATTER);
+            LocalDate upper = LocalDate.parse(splittedRange[1].trim(), FORMATTER);
+            return new Range<LocalDate>(lower, upper);
+        } catch (DateTimeParseException e) {
+            throw new ParseException("Invalid Date Format: Please input dates in the format 'dd/MM/yyyy'");
+        }
+    }
+
+    /**
+     * Parses a range in string into Range object with type Integer.
+     *
+     * @param trimmedRange the string representing the age range.
+     * @return a {@code Range<Integer>} object containing the parsed start and end dates.
+     * @throws ParseException if the input string is not in the correct format or not a valid integer.
+     */
+    private Range<Integer> parseAgeRange(String trimmedRange) throws ParseException {
+        String[] splittedRange = trimmedRange.split("-");
+
+        if (splittedRange.length != 2) {
+            throw new ParseException("Incorrect range format: Please provide your age range as "
+                   + "ageValue - ageValue");
+        }
+
+        try {
+            int lower = Integer.parseInt(splittedRange[0].trim());
+            int upper = Integer.parseInt(splittedRange[1].trim());
+            return new Range<Integer>(lower, upper);
+        } catch (NumberFormatException e) {
+            throw new ParseException("Invalid Age Format: Please provide a valid age value");
+        }
+    }
+
+    /**
+     * Returns true if at least one of the prefixes contains values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean isPrefixPresent(ArgumentMultimap argumentMultimap, Prefix prefix) {
+        return argumentMultimap.getValue(prefix).isPresent();
+    }
+
+}
```

Comment:
Maybe FilterCommandParser could have a FilterCommandParserTest?

Reply:
yes! it is stated as my coming implementation on my pr description